File: src/modules/app/api/settings/entities/employee.ts
import { Entity } from '@/core';

/**
 * Employee of a business that can perform services.
 */
export class Employee extends Entity {
    public static NAME_MAX_LENGTH = 64;
    public static POSITION_MAX_LENGTH = 32;

    /**
     * Create a new employee entity.
     * @param name The employee's full name.
     * @param position Their job title.
     */
    constructor(public name: string, public position: string = '') {
        super();

        if (name.length > Employee.NAME_MAX_LENGTH) {
            throw new RangeError('name');
        }

        if (position.length > Employee.POSITION_MAX_LENGTH) {
            throw new RangeError('position');
        }
    }
}

File: backend/src/Domain/Settings/Entities/Employee.cs
using System;

namespace DetailingArsenal.Domain {
    public class Employee : Entity<Employee>, IUserEntity {
        public const int NameMaxLength = 64;
        public const int PositionMaxLength = 32;

        public Guid UserId { get; set; } = Guid.Empty;

        /// <summary>
        /// Legal, or nickname of the employee.
        /// </summary>
        public string Name {
            get => name;
            set {
                if (value.Length > NameMaxLength) {
                    throw new ArgumentOutOfRangeException();
                }

                name = value;
            }
        }

        /// <summary>
        /// The title of their job position.
        /// </summary>
        public string? Position {
            get => position;
            set {
                if (value?.Length > PositionMaxLength) {
                    throw new ArgumentOutOfRangeException();
                }

                position = value;
            }
        }

        private string name = null!;
        private string? position = null;
    }
}